File: sText/src/main/java/com/samid/sTextView/base/STextView.java
package com.samid.sTextView.base;

import android.content.Context;
import android.util.AttributeSet;

public abstract class STextView extends androidx.appcompat.widget.AppCompatTextView {
    public STextView(Context context) {
        this(context, null);
    }

    public STextView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public STextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public abstract void setAnimationListener(AnimationListener listener);

    public abstract void setProgress(float progress);

    public abstract void animateText(CharSequence text);
}

File: sText/src/main/java/com/samid/sTextView/base/STextView.kt
package com.samid.stextview.base

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import com.samid.stextview.base.AnimationListener

abstract class STextView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : AppCompatTextView(context, attrs, defStyleAttr) {

    abstract fun setAnimationListener(listener: AnimationListener)
    abstract fun setProgress(progress: Float)
    abstract fun animateText(text: CharSequence)
}